services/sonar/sonar-base.js

```diff
@@ -83,12 +84,22 @@ module.exports = class SonarBase extends BaseJsonService {
     const metrics = {}
 
     if (useLegacyApi) {
-      json[0].msr.forEach(measure => {
+      const [{ msr: measures }] = json
+      if (!measures.length) {
+        throw new NotFound({ prettyMessage: 'metric not found' })
+      }
```

Comment:
It turns out that the API (both current and legacy) will return a 404 for an unsupported metric, but it can also return an http 200 with an empty array if the metric is valid, but the target component does not publish the corresponding data (for example if the project is not publishing test results). This also happens sometimes when the metric key was previously valid, but has since been deprecated

Reply:
Is it worth adding that in a comment, too?

Reply:
😄 it'd definitely be more useful in the long term to be inline the code instead of an aside in a PR wouldn't it.

updating that now